pkg/operator/controller/controller_router_deployment_test.go

```diff
@@ -201,6 +209,11 @@ func TestDesiredRouterDeployment(t *testing.T) {
 		t.Errorf("router Deployment has unexpected node selector: %#v",
 			deployment.Spec.Template.Spec.NodeSelector)
 	}
+	if len(deployment.Spec.Template.Spec.Tolerations) != 1 ||
+		deployment.Spec.Template.Spec.Tolerations[0] != ci.Spec.NodePlacement.Tolerations[0] {
+		t.Errorf("router Deployment has unexpected toleration: %#v",
+			deployment.Spec.Template.Spec.Tolerations[0])
```

Comment:
> We should probably have a check earlier, before we set ci.Spec.NodePlacement, to make sure that len(deployment.Spec.Template.Spec.Tolerations) is 0 in that case.

Done
> Inside this if for this check, we know that len(deployment.Spec.Template.Spec.Tolerations) is not 1

I'm not following... `len(deployment.Spec.Template.Spec.Tolerations)` is `1` since it was set earlier in the test: `Tolerations: []corev1.Toleration{toleration},`. Am I missing something in your comment?

Reply:
> I'm not following... `len(deployment.Spec.Template.Spec.Tolerations)` is `1` since it was set earlier in the test: `Tolerations: []corev1.Toleration{toleration},`. Am I missing something in your comment?

We set `ci.Spec.NodePlacement.Tolerations`.  Now we are checking what `desiredRouterDeployment` did with that input.  The `if` here is checking that `desiredRouterDeployment` copied `ci.Spec.NodePlacement.Tolerations` to `deployment.Spec.Template.Spec.Tolerations`.  If not, then we need to print an error message, but that error message cannot make assumptions about what `desiredRouterDeployment` actually assigned to `deployment.Spec.Template.Spec.Tolerations`.

Reply:
I understand now. Thank you for the detailed explanation.